mlir/lib/Conversion/GPUCommon/OpToFuncCallLowering.h

```diff
@@ -89,7 +91,14 @@ struct OpToFuncCallLowering : public ConvertOpToLLVMPattern<SourceOp> {
 private:
   Value maybeCast(Value operand, PatternRewriter &rewriter) const {
     Type type = operand.getType();
-    if (!isa<Float16Type>(type))
+    if (!isa<FloatType>(type))
+      return operand;
+
+    // if there's a f16 function, no need to cast f16 values
+    if (!f16Func.empty() && isa<Float16Type>(type))
+      return operand;
+
+    if (isa<Float64Type>(type) || isa<Float32Type>(type))
```

Comment:
wont this also cast for other non f16/f32/f64 types?
I think you just want to modify the exisiting check to 
```
    // if there's a f16 function, no need to cast f16 values
    if (!isa<Float16Type>(type) || !f16Func.empty())
      return operand;
 ```
and leave rest of the logic as is?

Reply:
But we do want to cast, for, say, bf16, I think

Or at least it feels reasonable to do that.

Also, nit, combine the `isa<>`s together

Reply:
That would be different functionality then what the function currently does but yes seems reasonable to do, probably want to add a test for it though. showing the casting happening for bf16 for example.

Reply:
Also this comment will need updating
https://github.com/llvm/llvm-project/blob/main/mlir/lib/Conversion/GPUCommon/OpToFuncCallLowering.h#L23

Reply:
Hi @nirvedhmeshram, the comment is already updated. Thanks for your suggestion, I'll add some tests.